tests/unit/test_solvers.py

```diff
@@ -524,6 +524,315 @@ def test_herding_state(self, reduce_problem: _ReduceProblem) -> None:
         expected_state = HerdingState(solver.kernel.gramian_row_mean(dataset))
         assert eqx.tree_equal(state, expected_state)
 
+    def test_kernel_herding_analytic_unique(self) -> None:
+        # pylint: disable=line-too-long
+        r"""
+        Test kernel herding on an analytical example, enforcing a unique coreset.
+
+        In this example, we have data of:
```

Comment:
Firstly, this is such a good example and explanation and makes KernelHerding so easy to understand. Thank you for adding it :)
`In this example, we have data of:` can become
`In this example, we have a data of:`

Reply:
Thanks - I think using data without the a preceding it is what we do in the rest of the codebase but happy to be changed if we don't